src/main/java/seedu/storage/IngredientList.java

```diff
@@ -1,33 +1,57 @@
 package seedu.storage;
 
 import seedu.data.ingredient.Ingredient;
+
 import java.util.ArrayList;
 
 public class IngredientList {
     protected ArrayList<Ingredient> ingredientList;
 
     public IngredientList() {
-        ingredientList = new ArrayList<Ingredient>();
+        this.ingredientList = new ArrayList<Ingredient>();
     }
 
     /**
      * Adds an ingredient to ingredientList.
+     *
      * @param ingredient The Ingredient to be added.
      */
     public void addIngredient(Ingredient ingredient) {
-        ingredientList.add(ingredient);
+        this.ingredientList.add(ingredient);
+    }
+
+    /**
+     * Sorts the ingredients in ingredientList by date if byDate is true and by description if false.
+     *
+     * @param byDate toggle to sort by name or expiry.
+     * @return sorted ingredient ArrayList.
+     */
+    public ArrayList<Ingredient> sortIngredient(boolean byDate) {
+        ArrayList<Ingredient> sortedList = new ArrayList<Ingredient>(ingredientList);
+        sortedList.sort((byDate ? Ingredient.IngExpiryComparator : Ingredient.IngNameComparator));
+        return sortedList;
     }
 
     /**
      * Returns the current list of ingredients.
+     *
      * @return List of ingredients.
      */
-    public ArrayList<Ingredient> getIngredientList() {
-        return ingredientList;
+    public ArrayList<Ingredient> getIngredientList(String sortType) {
+        switch (sortType.toLowerCase()) {
+        case "e":
+            return sortIngredient(true);
+        case "default":
+            return sortIngredient(false);
+        default:
+            return ingredientList;
+        /*TODO default throw exception*/
```

Comment:
Can remove this comment also, I think there is no exception to throw because default is to list based on how recent the ingredient is added right?

Reply:
this exception captures things like when user inputs unrecognisable sort type. Like `list --` or `list -p` then we have to catch and ask them input in correct format

Reply:
oh I see